src/cascadia/TerminalSettingsEditor/MainPage.cpp

```diff
@@ -64,69 +64,95 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
 
         co_await winrt::resume_foreground(Dispatcher());
 
-        // "remove" all profile-related NavViewItems
-        // LOAD-BEARING: use Visibility here, instead of menuItems.Remove().
-        //               Remove() works fine on NavViewItems with an hstring tag,
-        //               but causes an out-of-bounds error with Profile tagged items.
-        //               The cause of this error is unknown.
+        // Deduce information about the currently selected item
+        std::optional<IInspectable> selectedItemTag{ std::nullopt };
+        std::optional<uint32_t> selectedItemIndex{ std::nullopt };
         auto menuItems{ SettingsNav().MenuItems() };
-        for (auto i = menuItems.Size() - 1; i > 0; --i)
+        if (const auto selectedItem{ SettingsNav().SelectedItem() })
         {
-            if (const auto navViewItem{ menuItems.GetAt(i).try_as<MUX::Controls::NavigationViewItem>() })
+            if (const auto navViewItem{ selectedItem.try_as<MUX::Controls::NavigationViewItem>() })
+            {
+                selectedItemTag = navViewItem.Tag();
+            }
+            menuItems.IndexOf(selectedItem, *selectedItemIndex);
+        }
+
+        // remove all profile-related NavViewItems by populating a std::vector
+        // with the ones we want to keep.
+        // NOTE: menuItems.Remove() causes an out-of-bounds crash. Using ReplaceAll()
+        //       gets around this crash.
+        std::vector<IInspectable> menuItemsSTL;
+        for (uint32_t i = 0; i < menuItems.Size(); ++i)
+        {
+            const auto item{ menuItems.GetAt(i) };
+            if (const auto navViewItem{ item.try_as<MUX::Controls::NavigationViewItem>() })
             {
                 if (const auto tag{ navViewItem.Tag() })
                 {
                     if (tag.try_as<Editor::ProfileViewModel>())
                     {
-                        // hide NavViewItem pointing to a Profile
-                        navViewItem.Visibility(Visibility::Collapsed);
+                        // don't add NavViewItem pointing to a Profile
+                        continue;
                     }
                     else if (const auto stringTag{ tag.try_as<hstring>() })
                     {
                         if (stringTag == addProfileTag)
                         {
-                            // hide NavViewItem pointing to "Add Profile"
-                            navViewItem.Visibility(Visibility::Collapsed);
+                            // don't add the "Add Profile" item
+                            continue;
                         }
                     }
                 }
             }
+            menuItemsSTL.push_back(item);

```

Comment:
`emplace_back(std::move(item))` to conserve the ref count so you don't need to keep incrementing and decrementing it. remove the `const` on item to make it work.